rama-dns/src/chain.rs

```diff
@@ -0,0 +1,236 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use rama_net::address::Domain;
+
+use crate::{DenyAllDns, DnsResolver, InMemoryDns};
+
+#[derive(Debug)]
+pub struct DnsChainDomainResolveErr<E: 'static> {
```

Comment:
```suggestion
pub struct DnsChainDomainResolveErr<E> {
```

Reply:
Doesn't E need to be static to ensure there's no case like below ?
```rust
struct NonStaticError<'a> {
    message: &'a str  
}

let message = String::from("temp error");
let err = DnsChainDomainResolveErr {
    errors: vec![NonStaticError {
        message: &message  // The error exists after message be destroyed
    }]
};
```

Reply:
Yes more or less, but in general it's better not to add such bounds at the definition level, but add such bounds only the impl blocks where you really need them. Otherwise you are forced to add these bounds constantly. And in some cases it's even possible that such bounds are optional and can also work in some scenarios without them.